toolchains/xslt-M4/schema-gen/make-json-schema-metamap.xsl

```diff
@@ -194,6 +194,16 @@
             <xsl:apply-templates select="$decl/constraint/allowed-values"/>
         </xsl:variable>
```

Comment:
This is nice! but line 196 could be

```
<xsl:variable name="allow-other" select="$enumerations/@allow-other = 'yes'"/>
```

😎 

Maybe we want to note also in a comment a link to where it says what the default is (if no value is given)?

Reply:
Are `'yes'` `true()` equivalent? (In other languages they call this truthiness I am not sure how to look this up in the XSLT 2.x/3.x specs.) I just wanted to be very explicit. I know your approach also works and started with that, I thought it would be best to be explicit here.

Reply:
No, 'yes' doesn't go 'true()' as can be tested by making the cast explicit: `boolean('yes')` returns `true()` but so does `boolean('no')`. The casting rule (from XPath 1.0 in this case) is that among strings, only `''` (the empty or null string) returns `false()`.

Reply:
Based on the feedback from @wendellpiez, how about the following?
```suggestion
        </xsl:variable>
        <xsl:variable name="allow-other" select="$enumerations/@allow-other = 'yes'"/>
        <xsl:variable name="any-or-all" select="if ($allow-other) then 'anyOf' else 'allOf'"/> 
```

Reply:
It's understood with these reviews, is it not? that we also are putting those unit tests to confirm the logic is correct, not just details of syntax? I mean, I like what I'm seeing but I haven't tested it.  🤔